x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -29,3 +39,59 @@ export const validateEql = async ({
     signal,
   });
 };
+
+interface AggsParams extends EqlValidationRequest {
+  data: DataPublicPluginStart;
+  interval: Unit;
+  fromTime: string;
+  toTime: string;
+  signal: AbortSignal;
+}
+
+export const getEqlPreview = async ({
+  data,
+  index,
+  interval,
+  query,
+  fromTime,
+  toTime,
+  signal,
+}: AggsParams): Promise<EqlPreviewResponse> => {
+  try {
+    const response = await data.search
+      .search<EqlSearchStrategyRequest, EqlSearchStrategyResponse<EqlSearchResponse<Source>>>(
+        {
+          params: {
+            // @ts-expect-error allow_no_indices is missing on EqlSearch
+            allow_no_indices: true,
+            index: index.join(),
+            body: {
+              filter: {
+                range: {
+                  '@timestamp': {
+                    gte: toTime,
+                    lte: fromTime,
```

Comment:
As discussed over chat we need a "format" here for the ranges.

Reply:
Updated! Thanks for providing some context on that field for me.